### PR TITLE
Allow login and port to be specified when using `tsh config` to generate openssh configs

### PR DESCRIPTION
**Pull request by lxea**
Fix for #24234

### CODE REVIEW

**Review by r0mant — COMMENTED** [review]
Overall lgtm but one of the [comments](https://github.com/gravitational/teleport/issues/24234#issuecomment-1599264244) on the ticket also mentions that users would like to be able to control port in a similar way. Shall we add this as well while you're at it?

**Comment by lxea** [comment]
> Overall lgtm but one of the [comments](https://github.com/gravitational/teleport/issues/24234#issuecomment-1599264244) on the ticket also mentions that users would like to be able to control port in a similar way. Shall we add this as well while you're at it?

Ooh I missed that, is it the port [here](https://github.com/gravitational/teleport/pull/28994/files#diff-41796715f835e1344e1ca906d605b121c35580119d427a30219bff67701bb3e4R53) that should be configured or should it be in the other section

**Comment by r0mant** [comment]
@lxea Yes, that one.

File: lib/config/openssh/openssh.go [hunk 1]
```diff
@@ -68,7 +75,9 @@ type SSHConfigParameters struct {
 	ProxyHost           string
 	ProxyPort           string
 	ExecutablePath      string
+	Username            string
 	DestinationDir      string
+	Port                int
```
Comment (strideynet) [inline lib/config/openssh/openssh.go:80]: I feel like we might want to explain that `Port` here is the default port that will be used if no other port is specified. I can see this getting confusing with `ProxyPort` in the struxt as well.

**Review by strideynet — APPROVED** [review]

**Review by r0mant — APPROVED** [review]
lgtm but can you update the docs [reference](https://goteleport.com/docs/reference/cli/#tsh-config) to describe the relevant flags as well?

And let's backport this to v13 and v12.

**Pull request merged**